packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -217,25 +227,20 @@ class AndroidStudio implements Comparable<AndroidStudio> {
 
   List<String> get validationMessages => _validationMessages;
 
-  @override
-  int compareTo(AndroidStudio other) {
-    final int result = version.compareTo(other.version);
-    if (result == 0) {
-      return directory.compareTo(other.directory);
-    }
-    return result;
-  }
-
   /// Locates the newest, valid version of Android Studio.
+  ///
+  /// In the case that `--android-studio-dir` is configured, the version of
+  /// Android Studio found at that location is always returned, even if it is
+  /// invalid.
   static AndroidStudio? latestValid() {
-    final String? configuredStudio = globals.config.getValue('android-studio-dir') as String?;
-    if (configuredStudio != null) {
-      String configuredStudioPath = configuredStudio;
-      if (globals.platform.isMacOS && !configuredStudioPath.endsWith('Contents')) {
-        configuredStudioPath = globals.fs.path.join(configuredStudioPath, 'Contents');
+    final String? configuredStudioPath = globals.config.getValue('android-studio-dir') as String?;
+    if (configuredStudioPath != null) {
+      String correctedConfiguredStudioPath = configuredStudioPath;
+      if (globals.platform.isMacOS && !correctedConfiguredStudioPath.endsWith('Contents')) {
+        correctedConfiguredStudioPath = globals.fs.path.join(correctedConfiguredStudioPath, 'Contents');
       }
-      return AndroidStudio(configuredStudioPath,
-          configured: configuredStudio);
+      return AndroidStudio(correctedConfiguredStudioPath,
```

Comment:
I know we used to do this, but should we not toolexit in the case where `correctedConfiguredStudioPath` does not exist on disk? This most likely means they either made a mistake setting their config value, or forgot they set that and then deleted Android Studio.

Reply:
Added this and a test